src/Hal/Bundle/PhpMetricsCollector/Collector/PhpMetricsCollector.php

```diff
@@ -1,101 +1,122 @@
+        }
+        $classMetrics = array_filter($metrics->all(), function (Metric $metric) {
+            return $metric instanceof ClassMetric
+                || $metric instanceof InterfaceMetric;
+        });
+        $consolidated = new Consolidated($metrics);
+        $scoreByClass = [];
+        /** @var ClassMetric|InterfaceMetric $each */
+        foreach ($classMetrics as $each) {
+            $scoreByClass[$each->getName()] = [
+                'maintainability' => $each->get('mi'),
+                'commentWeight' => $each->get('commentWeight'),
+                'complexity' => $each->get('ccn'),
+                'loc' => $each->get('loc'),
+                'lloc' => $each->get('lloc'),
+                'cloc' => $each->get('cloc'),
+                'bugs' => $each->get('bugs'),
+                'difficulty' => $each->get('difficulty'),
+                'intelligentContent' => $each->get('intelligentContent'),
+                'vocabulary' => $each->get('vocabulary'),
+            ];
         }
 
         // average
-        if ($all['cfiles'] > 0) {
-            $average['maintainability'] = array_sum($all['maintainability']) / sizeof($all['maintainability']);
-            $average['commentWeight'] = array_sum($all['commentWeight']) / sizeof($all['commentWeight']);
-            $average['complexity'] = array_sum($all['complexity']) / sizeof($all['complexity']);
-            $average['loc'] = array_sum($all['loc']);
-            $average['lloc'] = array_sum($all['lloc']);
-            $average['cloc'] = array_sum($all['cloc']);
-            $average['bugs'] = array_sum($all['bugs']) / sizeof($all['bugs']);
-            $average['difficulty'] = array_sum($all['difficulty']) / sizeof($all['difficulty']);
-            $average['intelligentContent'] = array_sum($all['intelligentContent']) / sizeof($all['intelligentContent']);
-            $average['vocabulary'] = array_sum($all['vocabulary']) / sizeof($all['vocabulary']);
+        $average = [];
+        if (count($classMetrics) > 0) {
+            $calculateAvg = function ($property) use ($classMetrics) {
+                return round(array_sum(array_map(function (Metric $classMetric) use ($property) {
+                    return $classMetric->get($property);
+                }, $classMetrics)) / count($classMetrics), 2);
+            };
+            $average['maintainability'] = $consolidated->getAvg()->mi;
+            $average['commentWeight'] = $consolidated->getAvg()->commentWeight;
+            $average['complexity'] = $consolidated->getAvg()->ccn;
+            $average['loc'] = $calculateAvg('loc');
+            $average['lloc'] = $calculateAvg('lloc');
+            $average['cloc'] = $calculateAvg('cloc');
+            $average['bugs'] = $consolidated->getAvg()->bugs;
+            $average['difficulty'] = $consolidated->getAvg()->difficulty;
+            $average['intelligentContent'] = $consolidated->getAvg()->intelligentContent;
```

Comment:
`$consolidated->getAvg()` might hide calculations behind, it could be safer to keep it on a local variable.

```suggestion
            $averageConsolidated = $consolidated->getAvg();
            $average['maintainability'] = $averageConsolidated->mi;
            $average['commentWeight'] = $averageConsolidated->commentWeight;
            $average['complexity'] = $averageConsolidated->ccn;
            $average['loc'] = $calculateAvg('loc');
            $average['lloc'] = $calculateAvg('lloc');
            $average['cloc'] = $calculateAvg('cloc');
            $average['bugs'] = $averageConsolidated->bugs;
            $average['difficulty'] = $averageConsolidated->difficulty;
            $average['intelligentContent'] = $averageConsolidated->intelligentContent;
```